From fe0ccdfdfeb48e687743296988b2067a6c916900 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Sun, 28 Sep 2025 16:15:42 +0200
Subject: [PATCH 116/133] musl: delete dead __restore and __restore_rt
 functions for hexagon, loongarch64, riscv32, riscv64
X-Developer-Signature: v=1; a=openpgp-sha256; l=3860; i=xtex@astrafall.org;
 h=from:subject; bh=jejjGLZ+yskzmb33z96x3W0cVEyl77qHyHF5rKis0uI=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78nhn/bksdWeXe+kBT40EtkWPdeJNlU7l14T8b63X2
 sW7//rvjoksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICpLj3HyPBSLfLB1uzO+2W8vmJqd05NmLl+g/+fORIvXtjP5pvseukqI8MTgSK
 zyU+2HNaQ0ojY/283K/cLpnmuyzSfld5O55ryaB83AA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

https://www.openwall.com/lists/musl/2025/09/28/2
(cherry picked from commit ba19c1104bab8cea50eff7fbdf3b04ca33891d0b)
---
 lib/libc/musl/src/signal/hexagon/restore.s     | 11 -----------
 lib/libc/musl/src/signal/loongarch64/restore.s | 10 ----------
 lib/libc/musl/src/signal/riscv32/restore.s     | 10 ----------
 lib/libc/musl/src/signal/riscv64/restore.s     | 10 ----------
 src/libs/musl.zig                              |  4 ----
 5 files changed, 45 deletions(-)
 delete mode 100644 lib/libc/musl/src/signal/hexagon/restore.s
 delete mode 100644 lib/libc/musl/src/signal/loongarch64/restore.s
 delete mode 100644 lib/libc/musl/src/signal/riscv32/restore.s
 delete mode 100644 lib/libc/musl/src/signal/riscv64/restore.s

diff --git a/lib/libc/musl/src/signal/hexagon/restore.s b/lib/libc/musl/src/signal/hexagon/restore.s
deleted file mode 100644
index f43f5e02dc0e..000000000000
--- a/lib/libc/musl/src/signal/hexagon/restore.s
+++ /dev/null
@@ -1,11 +0,0 @@
-// TODO - Test this if sa_restorer is ever supported in our kernel
-.global __restore
-.type __restore,%function
-.global __restore_rt
-.type __restore_rt,%function
-__restore:
-__restore_rt:
-	r6 = #139				// SYS_rt_sigreturn
-	trap0(#0)
-.size __restore, .-__restore
-.size __restore_rt, .-__restore_rt
diff --git a/lib/libc/musl/src/signal/loongarch64/restore.s b/lib/libc/musl/src/signal/loongarch64/restore.s
deleted file mode 100644
index d90a8ebbdac9..000000000000
--- a/lib/libc/musl/src/signal/loongarch64/restore.s
+++ /dev/null
@@ -1,10 +0,0 @@
-.global __restore_rt
-.global __restore
-.hidden __restore_rt
-.hidden __restore
-.type   __restore_rt,@function
-.type   __restore,@function
-__restore_rt:
-__restore:
-	li.w    $a7, 139
-	syscall 0
diff --git a/lib/libc/musl/src/signal/riscv32/restore.s b/lib/libc/musl/src/signal/riscv32/restore.s
deleted file mode 100644
index 5a0af6959d21..000000000000
--- a/lib/libc/musl/src/signal/riscv32/restore.s
+++ /dev/null
@@ -1,10 +0,0 @@
-.global __restore
-.hidden __restore
-.type __restore, %function
-__restore:
-.global __restore_rt
-.hidden __restore_rt
-.type __restore_rt, %function
-__restore_rt:
-	li a7, 139 # SYS_rt_sigreturn
-	ecall
diff --git a/lib/libc/musl/src/signal/riscv64/restore.s b/lib/libc/musl/src/signal/riscv64/restore.s
deleted file mode 100644
index 5a0af6959d21..000000000000
--- a/lib/libc/musl/src/signal/riscv64/restore.s
+++ /dev/null
@@ -1,10 +0,0 @@
-.global __restore
-.hidden __restore
-.type __restore, %function
-__restore:
-.global __restore_rt
-.hidden __restore_rt
-.type __restore_rt, %function
-__restore_rt:
-	li a7, 139 # SYS_rt_sigreturn
-	ecall
diff --git a/src/libs/musl.zig b/src/libs/musl.zig
index be7b0502f76c..a0e1d8c228d4 100644
--- a/src/libs/musl.zig
+++ b/src/libs/musl.zig
@@ -1571,13 +1571,11 @@ const src_files = [_][]const u8{
     "musl/src/signal/arm/sigsetjmp.s",
     "musl/src/signal/block.c",
     "musl/src/signal/getitimer.c",
-    "musl/src/signal/hexagon/restore.s",
     "musl/src/signal/hexagon/sigsetjmp.s",
     "musl/src/signal/i386/restore.s",
     "musl/src/signal/i386/sigsetjmp.s",
     "musl/src/signal/kill.c",
     "musl/src/signal/killpg.c",
-    "musl/src/signal/loongarch64/restore.s",
     "musl/src/signal/loongarch64/sigsetjmp.s",
     "musl/src/signal/m68k/sigsetjmp.s",
     "musl/src/signal/mips64/sigsetjmp.s",
@@ -1591,9 +1589,7 @@ const src_files = [_][]const u8{
     "musl/src/signal/psignal.c",
     "musl/src/signal/raise.c",
     "musl/src/signal/restore.c",
-    "musl/src/signal/riscv32/restore.s",
     "musl/src/signal/riscv32/sigsetjmp.s",
-    "musl/src/signal/riscv64/restore.s",
     "musl/src/signal/riscv64/sigsetjmp.s",
     "musl/src/signal/s390x/restore.s",
     "musl/src/signal/s390x/sigsetjmp.s",
-- 
2.52.0

